pulp_container/app/serializers.py

```diff
@@ -608,7 +612,9 @@ def validate(self, data):
                     _("Only one of 'containerfile' and 'containerfile_artifact' may be specified.")
                 )
             data["containerfile_artifact"] = Artifact.init_and_validate(data.pop("containerfile"))
-        elif "containerfile_artifact" not in data:
+        elif "containerfile_artifact" in data:
```

Comment:
i don't think this should be called here, at this point the artifact was just created in memory but not saved. This should suffice https://github.com/pulp/pulp_container/pull/362/files#diff-31c44b2812e50f3171248db027f439e8e64248cb84067fc7c305de74cdb9a86fR744

Reply:
Actually i believe this is the codepath where it was supplied as an artifact_href by the user.

Reply:
ooooh, you're right!